Single Logout (#4247)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/guides/single-logout/main/index.md

excerpt: Configure Single Logout for your apps.
layout: Guides
---

<ApiLifecycle access="ie" /><br>
<ApiLifecycle access="ea" />

This guide discusses how to configure Single Logout (SLO) for your apps.

---

**Learning outcomes**

* Understand the purpose of Single Logout (SLO)
* Understand how to configure SLO for your app

**What you need**

* [Okta Developer Edition organization](https://developer.okta.com/signup)
* Existing SAML app and OpenID Connect (OIDC) app integrations to update for SLO. See [Create SAML app integrations](https://help.okta.com/okta_help.htm?id=ext_Apps_App_Integration_Wizard-saml) or [Create OIDC app integrations](https://help.okta.com/okta_help.htm?id=ext_Apps_App_Integration_Wizard-oidc) if you don’t have configured app integrations.
* The SLO feature enabled for your org. From the left navigation pane in the Admin Console, go to **Settings** > **Features**, locate the SLO feature and enable.

---

## Overview

The Single Logout (SLO) feature allows a user to sign out of an SLO participating app on their device and end their Okta session. The user is then automatically signed out of all other SLO participating apps on other devices.

> Okta currently supports SP-initiated SLO for third-party SAML 2.0 and OpenID Connect (OIDC) apps. When an end user clicks the sign-out button in your app, the app directs the browser to Okta while making an inbound logout request. This indicates to Okta that the user wants to sign out of the app. In response, Okta ends the user’s Okta session.

Reviewer: ```suggestion
Okta currently supports Service Provider-initiated (SP-initiated) SLO for third-party SAML 2.0 and OpenID Connect (OIDC) apps. When an end user clicks the sign-out button in your app, the app directs the browser to Okta while making an inbound logout request. This indicates to Okta that the user wants to sign out of the app. In response, Okta ends the user’s Okta session.
```

The multiple device SLO feature supports outbound logout requests (IdP-initiated SLO) after the SP app makes the SP-initiated inbound logout request to Okta. Okta sends outbound logout requests to any other apps participating in SLO that didn't initiate the logout. This applies only to the downstream apps where the user has previously established a session. Requests are communicated from Okta to apps using front-channel logout, which means that the browser does the communicating.

> SLO is especially useful in scenarios where users share computers or use public kiosks. A user may sign in to a computer portal, and then open one app, and then another, and another, and so on. The signing in of the user for each app happens behind the scenes. In a perfect world, when the user wants to sign out, they should sign out of every app to keep the next user from accessing their information. But, most users don’t do that. SLO logs the user out of everything at once.

Reviewer: ```suggestion
SLO is especially useful in scenarios where users share computers or use public kiosks. A user may sign in to a computer portal and then open multiple apps. The user sign in process for each app happens behind the scenes. In a perfect world, when the user is finished, they should sign out of every app to keep the next user from accessing their information. But, most users don’t do that. SLO logs the user out of everything at once.
```

### Single Logout diagram

<div class="three-quarter">

![Flow diagram that displays three event examples of multiple device single logout using three apps and including three Okta sessions](/img/slo-multiple-device.png)

</div>

**Event 1**

* The user signs out of App 1 using Browser 1.
* App 1 initiates the logout (SP-initiated) by sending a front-channel inbound logout request to Okta using Browser 1. For example:

    `GET https://${yourOktaDomain}/oauth2/v1/logout?id_token_hint=<idToken>&post_logout_redirect_uri=<configuredPostLogoutRedirectUri>&state=<someState>`

* Okta ends Okta Session 1. The user can still access Apps 2 and 3 within the scope of each app session.

**Event 2**

> * Okta determines other apps that the user was signed in to in Okta Session 1, which are Apps 2 and 3.

Reviewer: ```suggestion
* Okta determines that Apps 2 and 3 were also part of Okta Session 1.
```
